BuildingAndTestingInBunsen.md

```diff
@@ -0,0 +1,119 @@
+  * Otherwise, you'll need to look for the code in either repository.  If you're not sure, then you should likely start on the [vdo-devel](https://github.com/dm-vdo/vdo-devel) repository and move to the [common](https://github.com/dm-vdo/common) repository if you still couldn't find what you were looking for.
+
+# Find the code
+
+For the purposes of this guide, all of the work you will be doing is going to take place on the Bunsen `resource` system.
+
+SSH to the `resource` system (typically named `ossbunsen-resource`) as the user that you have provisioned it with.
+
+
+For changes to the [vdo-devel](https://github.com/dm-vdo/vdo-devel) repository:
+
+  * Starting in the home directory, clone the repository and cd into it
+    ```
+    git clone https://github.com/dm-vdo/vdo-devel.git && cd vdo-devel
+    ```
+  * The files under src from here can be generally distinguished by language-type (though `c++` isn't accurate).
+    * Core VDO code can be found under src/c++/vdo
+    * Perl test code can be found under src/perl/Permabit
+
+For changes to the [common](https://github.com/dm-vdo/common) repository:
+
+  * Starting in the home directory, clone the repository and cd into it
+    ```
+    git clone https://github.com/dm-vdo/common.git && cd common
+    ```
+  * This layout is similar to vdo-devel, except there is no src directory at the top level.
+    * Perl code which is used for testing and other support functions can be found under perl/
+    * Packaging code to generate the various RPMs needed for Bunsen can be found under packaging/
+
+# Editing
+
+Once you've cloned your tree to the resource machine, then you can go in and start making changes.
+
+  * If you are intending to post these changes to another git repository, you will want to add the relevant repositories as a remote in the repository.
+  ```
+  git remote add <name> <url>
+  ```
+  NOTE: Additional `git remote` operations can be studied through the usage of `git remote -h`.
+
+# Building
+
+The best way to build is to get into the base level of the repository and run `make`.
+
+Assuming the repository was cloned to your homedir:
+
+  * For [vdo-devel](https://github.com/dm-vdo/vdo-devel)
+
+    ```
+    cd ~/vdo-devel
+    make
+    ```
+
+  * For [common](https://github.com/dm-vdo/common)
+
+    ```
+    cd ~/common
+    make
+    ```
+
+# Testing
+
+As part of the checkin process, VDO's projects always undergo testing via Jenkins.  This testing is initiated by calling the `make jenkins` target.
+
+The `make jenkins` target is implemented in both the [common](https://github.com/dm-vdo/common) and [vdo-devel](https://github.com/dm-vdo/vdo-devel) projects.
+
+IMPORTANT: It all contributions made to these projects will run through the `make jenkins` target, so it is a good idea to run this up front to avoid any surprises.  It is generally the contributor's responsibility to ensure that all tests pass.
```

Comment:
"It all contributions" -- I think "It " is a typo.

"up front" - maybe "before submitting a pull request"?

Reply:
Right.  s/It all/All/ :)

and definitely yes for the second as well. :)

Reply:
Should be in the latest update contained within this PR.